controller/team.go

```diff
@@ -0,0 +1,103 @@
+package controller
+
+import (
+	"github.com/fabric8-services/fabric8-auth/app"
+	"github.com/fabric8-services/fabric8-auth/application"
+	"github.com/fabric8-services/fabric8-auth/authorization"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/jsonapi"
+	"github.com/fabric8-services/fabric8-auth/log"
+	"github.com/fabric8-services/fabric8-auth/login"
+	"github.com/goadesign/goa"
+	"strings"
+)
+
+// TeamController implements the team resource.
+type TeamController struct {
+	*goa.Controller
+	app application.Application
+}
+
+// NewTeamController creates a team controller.
+func NewTeamController(service *goa.Service, app application.Application) *TeamController {
+	return &TeamController{Controller: service.NewController("TeamController"), app: app}
+}
+
+// Create runs the create action.
+func (c *TeamController) Create(ctx *app.CreateTeamContext) error {
+	currentUser, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError(err.Error()))
+	}
+
+	if currentUser == nil {
```

Comment:
login.ContextIdentity() can't return both identity and err == nil. Just check the error. No need to check currentUser == nil (in other places either).